### PR TITLE
fix: When converting a URL to a string, do not add a `?` if there are no query string parameters

### PR DESCRIPTION
**Pull request by JakeChampion**
fixes https://github.com/fastly/js-compute-runtime/issues/484

### CODE REVIEW

File: c-dependencies/js-compute-runtime/rust-url/src/lib.rs [hunk 1]
```diff
@@ -36,13 +36,26 @@ impl JSUrlSearchParams {
         match self.url_or_str {
             UrlOrString::Url(url) => {
                 let url = unsafe { url.as_mut().unwrap() };
-                let mut pairs = url.url.query_pairs_mut();
-                pairs.clear().extend_pairs(self.list.iter());
+                match self.list.is_empty() {
+                    true => {
+                        url.url.set_query(None);
```
Comment (JakeChampion) [inline c-dependencies/js-compute-runtime/rust-url/src/lib.rs:41]: This is the crux of the bug-fix.
Previously we used `url.query_pairs_mut().clear()` however `url.query_pairs_mut().clear();` is equivalent to `url.set_query(Some(""))`, not `url.set_query(None)` and we really wanted `.set_query(None)`

**Review by guybedford — APPROVED** [review]

File: c-dependencies/js-compute-runtime/rust-url/src/lib.rs [hunk 2]
```diff
@@ -36,13 +36,26 @@ impl JSUrlSearchParams {
         match self.url_or_str {
             UrlOrString::Url(url) => {
                 let url = unsafe { url.as_mut().unwrap() };
-                let mut pairs = url.url.query_pairs_mut();
-                pairs.clear().extend_pairs(self.list.iter());
+                match self.list.is_empty() {
```
Comment (elliottt) [inline c-dependencies/js-compute-runtime/rust-url/src/lib.rs:39]: Could you switch this to an `if-then-else`, as it's matching a `bool` value?

**Review by elliottt — APPROVED** [review]
Looks good to me, great find!

**Comment by twhid** [comment]
Hi all -- is this live on fastly production services?

**Comment by JakeChampion** [comment]
> Hi all -- is this live on fastly production services?

Howdy, yes version 1.7.1 has this included 👍  https://github.com/fastly/js-compute-runtime/blob/main/CHANGELOG.md#171-2023-04-11

**Pull request merged**